src/main/java/org/fenixedu/academic/domain/person/personIdentifier/PersonIdentifier.java

```diff
@@ -0,0 +1,34 @@
+package org.fenixedu.academic.domain.person.personIdentifier;
+
+import java.util.Optional;
+
+import org.apache.commons.lang3.StringUtils;
+import org.fenixedu.academic.domain.exceptions.DomainException;
+import org.fenixedu.bennu.core.domain.Bennu;
+
+public class PersonIdentifier extends PersonIdentifier_Base {
+
+    public PersonIdentifier() {
```

Comment:
It's better to expose a create method, receiving a person and an identifier (and make this constructor protected)